test/integration/edition_edit_test.rb

```diff
@@ -35,4 +31,59 @@ class EditionEditTest < IntegrationTest
     assert page.has_text?("Related external links")
     assert page.has_text?("Unpublish")
   end
+
+  context "#metadata" do
+    context "#draft" do
+      setup do
+        click_link("Metadata")
+      end
+
+      should "'Metadata' header and an update button" do
+        within :css, ".gem-c-heading" do
+          assert page.has_text?("Metadata")
+        end
+        assert page.has_button?("Update")
+      end
+
+      should "show slug input box prefilled" do
+        assert page.has_text?("Slug")
+        assert page.has_text?("If you change the slug of a published page, the old slug will automatically redirect to the new one.")
+        assert page.has_field?("artefact[slug]", with: /slug/)
+      end
+
+      should "update and show success message" do
+        fill_in "artefact[slug]", with: "changed-slug"
+        choose("Welsh")
+        click_button("Update")
+
+        assert find(".gem-c-radio input[value='cy']").checked?
+        assert page.has_text?("Metadata has successfully updated")
+        assert page.has_field?("artefact[slug]", with: "changed-slug")
+      end
+    end
+
+    context "#published" do
+      setup do
+        edition = FactoryBot.create(
+          :completed_transaction_edition,
+          panopticon_id: FactoryBot.create(
+            :artefact,
+            slug: "can-i-get-a-driving-licence",
+          ).id,
+          state: "published",
+          slug: "can-i-get-a-driving-licence",
+        )
+
+        visit edition_path(edition)
+        click_link("Metadata")
+      end
+
+      should "show current value for slug and language as texts" do
```

Comment:
Not sure "as texts" is good English. We can probably just omit that bit anyway as it should be clear from reading the assertions that it is testing the displayed UI text.